libraries/ESP8266WiFi/src/ESP8266WiFiScan.h

```diff
@@ -34,7 +34,7 @@ class ESP8266WiFiScanClass {
 
     public:
 
-        int8_t scanNetworks(bool async = false, bool show_hidden = false, uint8 channel = 0, uint8* ssid = NULL);

```

Comment:
This is a breaking change.
What about leaving the former API without the default values ?
```cpp
+ int8_t scanNetworks(bool async = false, bool show_hidden = false, uint8 channel, uint8* ssid);
```

Reply:
Looks fishy, and indeed, it leads to an ambiguity error for the case of default ssid argument, because the compiler cannot distiguish the actual parameter for `bool passive` from `uint8_t channel`. Better break once, for good, in the breaking 3.0 release.